lib/aur-chroot

```diff
@@ -205,36 +211,73 @@ while read -r key _ value; do
     case $key=$value in
         Server=file://*)
             bindmounts_rw+=("${value#file://}") ;;
+        CacheDir=*)
+            # XXX: cachedir binding is a hairy subject
+            # see https://gitlab.archlinux.org/archlinux/devtools/-/blob/master/src/arch-nspawn.in?ref_type=heads#L86
+            bindmounts_rw+=("${value}") ;;
     esac
 done < <(pacman-conf --config "$pacman_conf")
 wait "$!"
 
 # create new container, required for update/build steps
-if (( create )); then
+if (( create )) && [[ ! -d $directory/root ]]; then
     # default to base-devel or multilib-devel, unless packages are
     # specified on the command-line.
     # available packages depend on the configured pacman configuration
     if (( $# )); then
         base_packages=("$@")
-
-    # XXX: use pacini to not process Include directives in pacman.conf
-    # (not supported by devtools)
     elif [[ $(pacini --section=multilib "$pacman_conf") ]] && [[ $machine == "x86_64" ]]; then
-        base_packages=('base-devel' 'multilib-devel')
+        base_packages=('base' 'base-devel' 'multilib-devel')
     else
-        base_packages=('base-devel')
+        base_packages=('base' 'base-devel')
     fi
 
-    # parent path is not created by mkarchroot (#371)
-    if [[ ! -d $directory ]]; then
-        # shellcheck disable=SC2086
-        $AUR_PACMAN_AUTH install -d "$directory" -m 755 -v
-    fi
+    # create directories for pacman
 
-    if [[ ! -d $directory/root ]]; then
-        # shellcheck disable=SC2086
-        $AUR_PACMAN_AUTH mkarchroot -C "$pacman_conf" -M "$makepkg_conf" "$directory"/root "${base_packages[@]}"
-    fi
+    $AUR_PACMAN_AUTH install -d "$directory"/root -m 755 -v
```

Comment:
There's so many AUR_PACMAN_AUTH calls here, maybe the entire script should run/rexec as root, similar to what devtools does.

Reply:
I tried, but it looks like I have to add a lot of envvar to preserve_env. Not sure if that's worth it.

Reply:
A possible approach is to not do it in the script (and limit the elevation tool to `sudo --preserve-env`), but rexec with `$AUR_PACMAN_AUTH env ... aur-chroot`. Thoughts?

Reply:
I think I'm missing something: whatever we do we need to keep track of the variables anyway and keep the list updated. I can't spot the difference between using `env` or `$AUR_PACMAN_AUTH` directly (in this case).

Reply:
I went and implemented the `sudo --preserve-env` and rexec with a list of variables. At least there is only one place where we need to keep track of that :)